runner/requester.go

```diff
@@ -388,18 +388,24 @@ func (b *Requester) runWorkers(wt load.WorkerTicker, p load.Pacer) error {
 						b.config.log.Debugw("Creating worker with ID: "+wID, "workerID", wID)
 					}
 
+					var streamInterceptor StreamInterceptor
+					if b.config.streamInterceptorProviderFunc != nil {
+						streamInterceptor = b.config.streamInterceptorProviderFunc()
+					}
+
 					w := Worker{
-						ticks:            ticks,
-						active:           true,
-						stub:             b.stubs[n],
-						mtd:              b.mtd,
-						config:           b.config,
-						stopCh:           make(chan bool),
-						workerID:         wID,
-						dataProvider:     b.dataProvider,
-						metadataProvider: b.metadataProvider,
-						streamRecv:       b.config.recvMsgFunc,
-						msgProvider:      b.config.dataStreamFunc,
+						ticks:             ticks,
+						active:            true,
+						stub:              b.stubs[n],
+						mtd:               b.mtd,
+						config:            b.config,
+						stopCh:            make(chan bool),
+						workerID:          wID,
+						dataProvider:      b.dataProvider,
+						metadataProvider:  b.metadataProvider,
+						streamRecv:        b.config.recvMsgFunc,
+						streamInterceptor: streamInterceptor,
```

Comment:
This looks like something shared by all the streams.
If that's the case, this won't work. The interceptor needs to be created when the stream is created.
E.g. when `w.stub.InvokeRpcBidiStream(*ctx, w.mtd, callOptions...)` is called.

Reply:
That makes sense. I've moved this so we create a new interceptor each time the worker makes a request.